tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for source host
+#
+# - Get resource data and dst host IP address
+# - Collect guest data, name, version, arch, image path, reusability ...
+# - Generate profile from autoyast template based on guest data
+# - First stage guest install
+# - Second stage guest install
+# - Get IP address of Guest and web page content. installation complete
+# - Set up passwordless ssh authentication
+# - Execute migration test cases
+# - Heath check guest on the dts host after migration
+# - clean up
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_src;
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use File::Copy 'copy';
+use File::Path 'make_path';
+use lockapi;
+use mmapi;
+use scheduler 'get_test_suite_data';
+
+my $vms            = {};
+my $vms_to_install = 0;
+my $images_path    = "/var/lib/libvirt/images";
+my $nfs_path       = get_var("NFS_PATH") ? get_var("NFS_PATH") : get_var('SUT_IP') . ":$images_path";
+my $guest          = get_required_var('GUEST');
+my $virt_insatll = "/usr/bin/virt-install --noautoconsole --network bridge=br0 --vcpus 2,maxvcpus=4 --memory 2048,maxmemory=4096 --events on_reboot=restart --video=vga --serial pty -x 'quiet textmode=1 sshd=1 sshpassword=novell'";
+my $sut_ip = get_var('SUT_IP');
+#my $inst_src = get_test_suite_data()->{location}->{"$guest"};
+my $inst_src;
+my $dst_ip;
+
+#collect parameters for each vm.
+sub init_vm_info {
```

Comment:
We already have `virt_autotest::utils::import_guest()`.

Reply:
import_guest uses NAT network. This test uses bridged network. That's quite different.   The way to get guest IP is different too. Please see https://confluence.suse.com/display/~tonyyuan/Live+migration+automation

Reply:
> Wow, this is very impressive work @tonyyuan1, thank you!
> 
> But this is also very big pull request, I suggest you to split it:
> The new universal autoyast profiles are very good, but I would introduce those in separate pull request.
> 
> But over all this is very impressive, thank you!

Thank you, Pavel.
Before starting this automation, I actually checked quite some exesting virt test modules, code in openqa to figure out what automation I want it to be. I set these goals: simplicity, low maintenance cost, rubost, easy to understand.
I used some different test approachs from virt test running in openqa. I will be very happy to discuss them with you, felix, Alice and others. We might exchange some views on how  good automation look like.

Reply:
Why is the bridged network necessary?

Reply:
bridged is commonly used in production by customers. bridged is easier to access to Guest whereas NAT reject all outside connection. 
I explained in https://confluence.suse.com/display/~tonyyuan/Live+migration+automation.

Reply:
In addition, if we want to extend and test qemu command options in the future bridged is the only way to go.

Reply:
No, NAT is used commonly as well. We have networking test for testing all the networking types already. Here, we test the migration, we already have a list of guests and their addresses and we access them only from the hypervisor.

Reply:
The page https://confluence.suse.com/display/~tonyyuan/Live+migration+automation does not exists/is not accessible anymore.

However I believe bridged networking is the better approach in general for the live migration test, because that's what is commonly used. When installing a KVM/XEN hypervisor, by default a bridged connection is used.

However we should have in mind to not create unnecessary code/data duplications, unless really necessary.